From: Gerd Hoffmann <kraxel@redhat.com>
Date: Wed, 26 Jul 2017 17:29:16 +0200
Subject: [PATCH] ui: update keymaps

Add recently added QKeyCodes to the keymaps.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Reviewed-by: Eric Blake <eblake@redhat.com>
Message-id: 20170726152918.11995-4-kraxel@redhat.com
(cherry picked from commit 9ade7759edbbdafa980ea974a629a8d193de5bc2)
---
 ui/input-keymap.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ui/input-keymap.c b/ui/input-keymap.c
index d4972bb364..7461e1edde 100644
--- a/ui/input-keymap.c
+++ b/ui/input-keymap.c
@@ -94,6 +94,9 @@ static int linux_to_qcode[KEY_CNT] = {
     [KEY_KPASTERISK]     = Q_KEY_CODE_KP_MULTIPLY,
     [KEY_F11]            = Q_KEY_CODE_F11,
     [KEY_F12]            = Q_KEY_CODE_F12,
+    [KEY_RO]             = Q_KEY_CODE_RO,
+    [KEY_HIRAGANA]       = Q_KEY_CODE_HIRAGANA,
+    [KEY_HENKAN]         = Q_KEY_CODE_HENKAN,
     [KEY_RIGHTCTRL]      = Q_KEY_CODE_CTRL_R,
     [KEY_SYSRQ]          = Q_KEY_CODE_SYSRQ,
     [KEY_RIGHTALT]       = Q_KEY_CODE_ALT_R,
@@ -107,6 +110,9 @@ static int linux_to_qcode[KEY_CNT] = {
     [KEY_PAGEDOWN]       = Q_KEY_CODE_PGDN,
     [KEY_INSERT]         = Q_KEY_CODE_INSERT,
     [KEY_DELETE]         = Q_KEY_CODE_DELETE,
+    [KEY_POWER]          = Q_KEY_CODE_POWER,
+    [KEY_KPCOMMA]        = Q_KEY_CODE_KP_COMMA,
+    [KEY_YEN]            = Q_KEY_CODE_YEN,
     [KEY_LEFTMETA]       = Q_KEY_CODE_META_L,
     [KEY_RIGHTMETA]      = Q_KEY_CODE_META_R,
     [KEY_MENU]           = Q_KEY_CODE_MENU,
@@ -242,6 +248,7 @@ static const int qcode_to_number[] = {
     [Q_KEY_CODE_RO] = 0x73,
     [Q_KEY_CODE_HIRAGANA] = 0x70,
     [Q_KEY_CODE_HENKAN] = 0x79,
+    [Q_KEY_CODE_POWER] = 0xde,
     [Q_KEY_CODE_YEN] = 0x7d,
     [Q_KEY_CODE_KP_COMMA] = 0x7e,
 
